precompiles/ArbNativeToken.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2021-2025, Offchain Labs, Inc.
+// For license information, see https://github.com/OffchainLabs/nitro/blob/master/LICENSE.md
+
+package precompiles
+
+import (
+	"errors"
+
+	"github.com/holiman/uint256"
+
+	"github.com/ethereum/go-ethereum/core/tracing"
+	"github.com/ethereum/go-ethereum/params"
+
+	"github.com/offchainlabs/nitro/util/arbmath"
+)
+
+// ArbNativeToken precompile enables minting and burning native tokens.
+// All calls to this precompile are authorized by the NativeTokenPrecompile wrapper.
+type ArbNativeToken struct {
+	Address addr
+}
+
+var mintBurnGasCost = arbmath.WordsForBytes(32) * params.SstoreSetGas / 100
```

Comment:
@diegoximenes, can you add a comment explaining why this is the right amount of gas to burn?

Reply:
I am not sure if this is the correct amount of gas to burn 😬.

`arbmath.WordsForBytes(32)` comes from the fact that `amount` in mint and burn has 32 bytes.
`params.SstoreSetGas` comes from the fact that I supposed that `evm.StateDB.AddBalance` would be the related inner operation.
`100` is inspired from [here](https://github.com/OffchainLabs/nitro/blob/358cb845d900d4c4ee669b117dad9c110df75858/precompiles/ArbRetryableTx.go#L166), I don't know why we use `100` there.

I was going to add a comment here about that, asking for opinions after marking the PR as ready to be reviewed 🙂

Reply:
* SSTORE stores 32Bytes, so no need to multiply by 32.
* The 100 where you pointed is for a keepalive, so it just "doesn't delete YET", which is why they take a fraction.

I currently think the best amount to charge is:
params.WarmStorageReadCostEIP2929 + params.CallValueTransferGas

This is based on CALL-cost:
The description is here:  https://github.com/wolflo/evm-opcodes/blob/main/gas.md#aa-call-operations
And a reasonable implementation here:
https://github.com/OffchainLabs/go-ethereum/blob/f40a18e3c41b032a628a5ef5dc8f9ae51369977f/core/vm/operations_acl_arbitrum.go#L107

We're doing a special case. Since we only mint and burn from the caller, we can assume this is always a warm access, and never a new account so it simplifies things.